src/Database/Transactions/GenericTransactionManager.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database\Transactions;
+
+use PDO;
+use Tempest\Database\Exceptions\TransactionException;
+use Throwable;
+
+final class GenericTransactionManager implements TransactionManager
```

Comment:
this is a bit oversimplified as it doesn't handle nested transactions, but it's a start :)